### PR TITLE
Implement SDT Module

### PR DESCRIPTION
**Pull request by RayLiu7**
# Description
Implement SDT Module
- Managing SDT on PowerFlex storage system:
  creating new SDT, getting details of SDT, managing IP/role of SDT,
  modifying attributes of SDT, and deleting SDT.

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, pep8, linting, or security issues
- [x] I have performed Ansible Sanity test using --docker default
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit Tests

### CODE REVIEW

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

File: plugins/modules/sdt.py [hunk 1]
```diff
@@ -0,0 +1,952 @@
...
+    description:
+    - State of the SDT.
+    choices: ['present', 'absent']
+    required: true
+    type: str
+  storage_port:
+    description:
+    - Storage port of the SDT.
+    type: int
+notes:
```
[the remark this answers is not in the document]
Comment (RayLiu7) [reply plugins/modules/sdt.py:82]: It supports check mode. Updated underneath.

File: plugins/modules/sdt.py [hunk 2]
```diff
@@ -0,0 +1,958 @@
...
+    - The name of the SDT.
+    - Mandatory for all operations.
+    - It is unique across the PowerFlex array.
+    required: true
+    type: str
+  sdt_new_name:
+    description:
+    - SDT new name, can only be used for renaming the SDT.
+    type: str
+  state:
```
Comment (trisha-dell) [inline plugins/modules/sdt.py:72]: state should be optional with default as present.

**Comment by sachin-apa** [comment]
run e2e

**Review by Bhavneet-Sharma — CHANGES_REQUESTED** [review]

File: plugins/modules/sdt.py [hunk 3]
```diff
@@ -0,0 +1,958 @@
...
+        )
+
+
+class SDTCreateHandler:
+    def handle(self, sdt_obj, sdt_params, sdt_details, protection_domain_id):
+        create_sdt_flag = False
+        sdt_ip_list = copy.deepcopy(sdt_params["sdt_ip_list"])
+        if sdt_params["state"] == "present" and not sdt_details:
+            if sdt_params["sdt_new_name"] or sdt_params["maintenance_mode"]:
+                error_msg = "sdt_new_name and maintenance_mode are not supported during SDT creation"
```
[the remark this answers is not in the document]
Comment (RayLiu7) [reply plugins/modules/sdt.py:915]: Hi @Bhavneet-Sharma, we had a agreement with Shrinidhi and Sachin to move modifying operations (like enter maintenance mode and rename) out of creation during the spike demo session, considering user usually have a sequence of steps to do in a playbook, so it's better to keep the atomicity of operations.

File: plugins/modules/sdt.py [hunk 4]
```diff
@@ -0,0 +1,958 @@
...
+        system"""
+        return Configuration(self.powerflex_conn, self.module).get_protection_domain(
+            protection_domain_name=protection_domain_name,
+            protection_domain_id=protection_domain_id,
+        )
+
+    def validate_create(self, protection_domain_id, sdt_ip_list, sdt_name):
+
+        if sdt_name is None or len(sdt_name.strip()) == 0:
+            error_msg = "Please provide valid sdt_name value for the creation of SDT."
```
[the remark this answers is not in the document]
Comment (RayLiu7) [reply plugins/modules/sdt.py:419]: Could you tell me why we don't want to use please in the msg? Just curious

File: plugins/modules/sdt.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Bhavneet-Sharma) [reply plugins/modules/sdt.py:915]: Hi @RayLiu7,
We had review comment from Anupam to not to through an error message when rename parameter is passed while creating the object.

File: plugins/modules/sdt.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Bhavneet-Sharma) [reply plugins/modules/sdt.py:419]: We had IDD review comment in the past about this. They said is not a best practice to add please error message.

File: plugins/modules/sdt.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (RayLiu7) [reply plugins/modules/sdt.py:915]: @sachin-apa @Bhavneet-Sharma ok, I'll remove the error display and ignore the params not applicable for create.

[the remark this answers is not in the document]
Comment (RayLiu7) [reply plugins/modules/sdt.py:915]: @sachin-apa Just to confirm, by ignoring the params not for create, we don't perform the rename or mode change in the this task, right?

File: plugins/modules/sdt.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (RayLiu7) [reply plugins/modules/sdt.py:90]: Added diff support and updated attributes

**Comment by RayLiu7** [comment]
Sonar and Checkmarx report
Sonar: https://sonarqube1.cec.lab.emc.com/dashboard?id=ansible-powerflex&branch=69
Checkmarx: https://osj-sio-03-prd.cec.lab.emc.com/job/Ansible-PowerFlex-CheckMarx/22/

File: plugins/modules/sdt.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (RayLiu7) [reply plugins/modules/sdt.py]: Fixed

File: plugins/modules/sdt.py [hunk 5]
```diff
@@ -0,0 +1,1017 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2024, Dell Technologies
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+""" Ansible module for managing SDT on Dell Technologies (Dell) PowerFlex"""
```
Comment (Bhavneet-Sharma) [inline plugins/modules/sdt.py:6]:
```suggestion
""" Ansible module for managing SDT on Dell Technologies PowerFlex"""
```

File: tests/unit/plugins/module_utils/mock_sdt_api.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
+# Copyright: (c) 2024, Dell Technologies
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+"""
+Mock Api response for Unit tests of SDT module on Dell Technologies (Dell) PowerFlex
```
Comment (Bhavneet-Sharma) [inline tests/unit/plugins/module_utils/mock_sdt_api.py:6]:
```suggestion
Mock Api response for Unit tests of SDT module on Dell Technologies PowerFlex
```

**Review by Bhavneet-Sharma — APPROVED** [review]

**Review by sachin-apa — APPROVED** [review]

**Review by trisha-dell — APPROVED** [review]

**Review by meenakshidembi691 — APPROVED** [review]

**Pull request merged**